packages/block-editor/src/components/iframe/index.js

```diff
@@ -306,13 +306,19 @@ function Iframe( {
 		iframeDocument.documentElement.classList.add( 'is-zoomed-out' );
 
 		const maxWidth = 750;
+		// This scaling calculation has to happen within the JS because CSS calc() can
+		// only divide and multiply by a unitless value. I.e. calc( 100px / 2 ) is valid
+		// but calc( 100px / 2px ) is not.
 		iframeDocument.documentElement.style.setProperty(
 			'--wp-block-editor-iframe-zoom-out-scale',
 			scale === 'default'
-				? Math.min( containerWidth, maxWidth ) /
+				? ( Math.min( containerWidth, maxWidth ) -
+						parseInt( frameSize ) * 2 ) /
```

Comment:
> Anyhow, my point is the is practically very similar yet at least in my PR the code acknowledges what’s inaccurate.

Yes - you figured out the main bug. That was a huge help. 🙏🏻 You were supposed to be credited in this PR because of that. This PR takes a slightly different direction by removing the border entirely and just account for the frame size scaling that you figured out. 

> This treats any frameSize value as a pixel. hat’s probably not a big problem or could even be argued that it creates some compensation for a non-pixel value even if it won’t be accurate. 

I'm fine forcing pixels for `frameSize`, as I don't think we can accurately account for non-pixel sizes in everything we need due to css calc limitations. I thought it was going to be possible to accurately account for any value, but now I'm convinced it isn't possible 😅 

> at least in my PR the code acknowledges what’s inaccurate.
[Do you mean this comment?](https://github.com/WordPress/gutenberg/pull/65814/files#diff-ab17abd38e687edd56573694ae3017e105e96b319571c190e9d4705c1872bc9cR310-R313). I tried [to explain it via this comment](https://github.com/WordPress/gutenberg/pull/65998/files#diff-ab17abd38e687edd56573694ae3017e105e96b319571c190e9d4705c1872bc9cR321), but maybe I didn't explain it in-depth enough to be useful.

Reply:
No worries about the credit. I do love that this ultimately does make the styles simpler. I thought I’d tried removing the border with my PR but seems like it was making it so the inline frame spacing wasn’t maintained through resizes.

> I thought it was going to be possible to accurately account for any value, but now I'm convinced it isn't possible 😅

Yeah, not without rendering another element to measure with `getComputedStyle` and some compromises. Trying to put the calculation in CSS was a great thought.

> but maybe I didn't explain it in-depth enough to be useful.

Oh, if that comment is meant to do that it seems misplaced. It’s detached from the JS calculation.

Reply:
> This treats any `frameSize` value as a pixel. That’s probably not a big problem or could even be argued that it creates some compensation for a non-pixel value even if it won’t be accurate. The approach in #65814 was different by avoiding factoring in a non-pixel `frameSize` but I think @ajlende mistook it for leaving out the `frameSize` entirely and called it breaking change [#65814 (comment)](https://github.com/WordPress/gutenberg/pull/65814#discussion_r1792131265). Then seemingly didn’t see my reply trying to clarify.

FWIW this approach also causes the breaking change that I commented on in your PR. I wasn't ready for this one to be merged so soon. I think still we need to document the requirement of a number representing pixels or a pixel string for the API of this component. Based on this argument.

> I suppose a loophole for the backwards compatibility argument could be that nothing about the prop was documented, so you could argue that the previous API accepted only numbers or pixel strings rather than any CSS unit.

And sorry for the slow response btw, I wanted to make sure that I understood your perspective and make sure the suggestions that I was about to share were valid and made sense which ended up taking all day for me yesterday. One thing that can annoy me is half-baked responses that seem dismissive or suggestions that won't work because they haven't fully been thought out, and I didn't want to do that to you. I have finally posted it https://github.com/WordPress/gutenberg/pull/65814/files#r1795825607, but it seems that ended up being too late.

Reply:
I am sorry I rushed merging this one! and also about not adding proper credit, I take full responsibility for that. I shouldn't have jumped on the merge button like that